### PR TITLE
ENHANCEMENT Remove jquery-ui button() api from default HTML editor dialog

### PR DESCRIPTION
**Pull request by tractorcow**
Necessary to fix CMS behat tests, now that we have merged the necessary changes into framework module.

Merge this before https://github.com/silverstripe/silverstripe-cms/pull/1699 to get those tests working again.

### CODE REVIEW

**Comment by tractorcow** [comment]
Note: Please do not merge until I've managed to confirm these fixes resolve the outstanding behat issues.

File: admin/client/src/legacy/HtmlEditorField.js [hunk 1]
```diff
@@ -991,10 +991,12 @@ $.entwine('ss', function($) {
...
       // Disable "insert" button if no files are selected
-      this.closest('ui-dialog')
-        .find('ui-dialog-buttonpane .media-insert')
-        .button(hasItems ? 'enable' : 'disable')
-        .toggleClass('ui-state-disabled', !hasItems);
+      this.closest('.ui-dialog')
+        .find('.ui-dialog-buttonpane .media-insert')
+        .toggleClass('ui-state-disabled', !hasItems)
+        .each(function() {
+          this.disabled = !hasItems;
```
Comment (flamerohr) [inline admin/client/src/legacy/HtmlEditorField.js]: `$(this).prop('disabled', !hasItems);` would be a safer

File: admin/client/src/legacy/HtmlEditorField.js [hunk 2]
```diff
@@ -1174,14 +1176,20 @@ $.entwine('ss', function($) {
...
       val = val.replace(/^https?:\/\//i, '');
-      if (orig !== val) this.val(val);
+      if (orig !== val) {
+        this.val(val);
+      }
 
-      this.getAddButton().button(!!val ? 'enable' : 'disable');
-      return !!val;
+      this.getAddButton().each(function() {
+        this.disabled = !hasVal;
```
Comment (flamerohr) [inline admin/client/src/legacy/HtmlEditorField.js]: same as above `$(this).prop('disabled', !hasVal);` would be a safer

**Comment by tractorcow** [comment]
ok, tests are passing; Once this is merged https://github.com/silverstripe/silverstripe-cms/pull/1699 should be repaired.

**Comment by chillu** [comment]
The broken tests on Travis are due to the fixed logic targeting https://github.com/silverstripe/silverstripe-cms/pull/1699. I've run some spot checks locally and looks good. Notably, this behaviour fixed the *old* "insert media" dialog which is only available when you don't have asset-admin installed.

**Pull request merged**